agenta-web/src/components/pages/evaluations/NewEvaluation/SelectEvaluatorSection.tsx

```diff
@@ -0,0 +1,184 @@
+
+type SelectEvaluatorSectionProps = {
+    evaluatorConfigs: EvaluatorConfig[]
+    evaluators: Evaluator[]
+    selectedEvalConfigs: string[]
+    setSelectedEvalConfigs: React.Dispatch<React.SetStateAction<string[]>>
+} & React.ComponentProps<typeof Collapse>
+
+const SelectEvaluatorSection = ({
+    evaluatorConfigs,
+    evaluators,
+    selectedEvalConfigs,
+    setSelectedEvalConfigs,
+    ...props
+}: SelectEvaluatorSectionProps) => {
+    const [searchTerm, setSearchTerm] = useState("")
+    const [isEvaluatorsModalOpen, setIsEvaluatorsModalOpen] = useState(false)
+    const [current, setCurrent] = useState(0)
+
+    const prevEvaluatorConfigsRef = useRef<EvaluatorConfig[]>(evaluatorConfigs)
+
+    useEffect(() => {
+        const prevConfigs = prevEvaluatorConfigsRef.current
+        const newConfigs = evaluatorConfigs.filter(
+            (config) => !prevConfigs.some((prevConfig) => prevConfig.id === config.id),
+        )
+
+        if (newConfigs.length > 0) {
+            setSelectedEvalConfigs((prevSelected) => [
+                ...prevSelected,
+                ...newConfigs.map((config) => config.id),
+            ])
+        }
+
+        prevEvaluatorConfigsRef.current = evaluatorConfigs
+    }, [evaluatorConfigs, setSelectedEvalConfigs])
+
+    const columns: ColumnsType<EvaluatorConfig> = [
+        // {
+        //     title: "Version",
+        //     dataIndex: "version",
+        //     key: "version",
+        //     onHeaderCell: () => ({
+        //         style: {minWidth: 80},
+        //     }),
+        // },
+        {
+            title: "Name",
+            dataIndex: "name",
+            key: "name",
+            render: (_, record) => {
+                return <div>{record.name}</div>
+            },
+        },
+        {
+            title: "Type",
+            dataIndex: "type",
+            key: "type",
+            render: (_, record) => {
+                const evaluator = evaluators.find((item) => item.key === record.evaluator_key)
+                return <Tag color={record.color}>{evaluator?.name}</Tag>
+            },
+        },
+    ]
+
+    const filteredEvalConfigs = useMemo(() => {
+        if (!searchTerm) return evaluatorConfigs
+        return evaluatorConfigs.filter((item) =>
+            item.name.toLowerCase().includes(searchTerm.toLowerCase()),
+        )
+    }, [searchTerm, evaluatorConfigs])
+
+    const selectedEvalConfig = useMemo(
+        () => evaluatorConfigs.filter((config) => selectedEvalConfigs.includes(config.id)),
+        [evaluatorConfigs, selectedEvalConfigs],
+    )
+
+    const handleRemoveEvalConfig = (configId: string) => {
+        const filterEvalConfig = selectedEvalConfigs.filter((id) => configId !== id)
+        setSelectedEvalConfigs(filterEvalConfig)
+    }
+
+    return (
+        <>
+            <Collapse
+                defaultActiveKey={["1"]}
+                {...props}
+                items={[
```

Comment:
let's improve this part. This is not good practice, because these'll be recreated each render. even if they don't change React will see the entire items prop as a new object and triggers unnecessary re-renders.